pkg/plugin/trans.go

```diff
@@ -152,6 +153,10 @@ func (r resourceTransition[T]) handleRequested(requested T, startingMigration bo
 			// adjust node pressure accordingly. We can have old < new or new > old, so we shouldn't
 			// directly += or -= (implicitly relying on overflow).
 			r.node.CapacityPressure = r.node.CapacityPressure - r.oldPod.capacityPressure + r.pod.CapacityPressure
+			if onlyThousands {
+				// this branch is for the case when caller requests increase in thousands
+				maxIncrease = T(math.Floor(float64(maxIncrease) / 1000.0))
+			}
```

Comment:
I believe this is incorrect: `maxIncrease` should just round down, not fully divide. Also, while it's probably not an issue here, float conversion is scary.

The placing here is also not _necessarily_ incorrect (provided `increase % 1000 == 0`, which caller must verify), but I feel it conveys intent better if it's outside the `if increase > maxIncrease`.

Here's what I'd recommend:

```diff
diff --git a/pkg/plugin/trans.go b/pkg/plugin/trans.go
index d745a6b..06f734d 100644
--- a/pkg/plugin/trans.go
+++ b/pkg/plugin/trans.go
@@ -148,6 +148,11 @@ func (r resourceTransition[T]) handleRequested(requested T, startingMigration bo
 		increase := requested - r.pod.Reserved
 		// Increases are bounded by what's left in the node
 		maxIncrease := remainingReservable
+		// if only in thousands, round down maxIncrease to nearest multiple of 1000
+		if onlyThousands {
+			thousand := T(100) * 10 // avoid compiler complaining about 1000 > maximum int8
+			maxIncrease = (maxIncrease / thousand) * thousand
+		}
 		if increase > maxIncrease /* increases are bound by what's left in the node */ {
 			r.pod.CapacityPressure = increase - maxIncrease
 			// adjust node pressure accordingly. We can have old < new or new > old, so we shouldn't
```